### PR TITLE
ARROW-11497: [Python] Provide parquet enable compliant nested type flag for python binding

### PR DESCRIPTION
**Pull request by trucnguyenlam**
Hi,

I am making a PR following the discussion in [ARROW-11497](https://issues.apache.org/jira/projects/ARROW/issues/ARROW-11497)

This is my first PR to this project, please let me know if I'm missing something, I will try to address all problem as much as I can.

Cheers,
Truc

### CODE REVIEW

**Comment by emkornfield** [comment]
@trucnguyenlam thank you for the PR it looks like the CI failure is legitimate here.  

Otherwise looks good to me

CC @jorisvandenbossche for python/cython style check

**Comment by trucnguyenlam** [comment]
@emkornfield thanks for your comment, I've also resolved python/cython style check, hopefully this should be good enough to be merged.

**Comment by trucnguyenlam** [comment]
Is it ok to be merged now? thanks

**Comment by trucnguyenlam** [comment]
@emkornfield @jorisvandenbossche Is it ok to be merged now? thanks

File: python/pyarrow/parquet.py [hunk 1]
```diff
@@ -541,6 +541,8 @@ def _sanitize_table(table, new_schema, flavor):
     The serialized Parquet data page format version to write, defaults to
     1.0. This does not impact the file schema logical types and Arrow to
     Parquet type casting behavior; for that use the "version" option.
+use_compliant_nested_type : bool, default False
+    Write compliant Parquet nested type, defaults to disable.
```
Comment (jorisvandenbossche) [inline python/pyarrow/parquet.py:545]: Can you expand the documentation a bit? Eg a user will wonder: what does "compliant" mean in this specific case, and what is different with the default of False?

**Comment by jorisvandenbossche** [comment]
@trucnguyenlam it seems you have some unrelated changes included (some git problem I suppose). Can you rebase this on master?

File: python/pyarrow/parquet.py (hunk 1 above)
Comment (trucnguyenlam) [reply python/pyarrow/parquet.py]: I've expanded documentation, please let me know if you are happy about it. :)

**Comment by trucnguyenlam** [comment]
@jorisvandenbossche done rebase and expanded doc :)

**Comment by trucnguyenlam** [comment]
@jorisvandenbossche please have a look if it is ok to merge? I resolved all issues as far as I know.

**Comment by trucnguyenlam** [comment]
@jorisvandenbossche could you please have a look at the PR? cheers

**Comment by trucnguyenlam** [comment]
@jorisvandenbossche @emkornfield this is open for quite some time, please have a look if this is eligible to be merged, thanks

**Comment by emkornfield** [comment]
+1, going to merge, we can address any items as a followup. @trucnguyenlam apologies for the delay.

**Pull request closed without merging**